perception/autoware_ground_segmentation/src/scan_ground_filter/node.cpp

```diff
@@ -176,30 +202,41 @@ void ScanGroundFilterComponent::convertPointcloud(
   const size_t in_cloud_data_size = in_cloud->data.size();
   const size_t in_cloud_point_step = in_cloud->point_step;
 
-  size_t point_index = 0;
-  pcl::PointXYZ input_point;
-  for (size_t global_offset = 0; global_offset + in_cloud_point_step <= in_cloud_data_size;
-       global_offset += in_cloud_point_step) {
-    // Point
-    get_point_from_global_offset(in_cloud, input_point, global_offset);
+  {  // add scope for time keeper
+    std::unique_ptr<ScopedTimeTrack> inner_st_ptr;
+    if (time_keeper_) inner_st_ptr = std::make_unique<ScopedTimeTrack>("scan", *time_keeper_);
+
+    size_t point_index = 0;
+    pcl::PointXYZ input_point;
+    for (size_t global_offset = 0; global_offset + in_cloud_point_step <= in_cloud_data_size;
+         global_offset += in_cloud_point_step) {
+      // Point
+      get_point_from_global_offset(in_cloud, input_point, global_offset);
 
-    auto radius{static_cast<float>(std::hypot(input_point.x, input_point.y))};
-    auto theta{normalizeRadian(std::atan2(input_point.x, input_point.y), 0.0)};
-    auto radial_div{static_cast<size_t>(std::floor(theta * inv_radial_divider_angle_rad))};
+      auto radius{static_cast<float>(std::hypot(input_point.x, input_point.y))};
+      auto theta{normalizeRadian(std::atan2(input_point.x, input_point.y), 0.0)};
+      auto radial_div{static_cast<size_t>(std::floor(theta * inv_radial_divider_angle_rad))};
 
-    current_point.radius = radius;
-    current_point.point_state = PointLabel::INIT;
-    current_point.orig_index = point_index;
+      current_point.radius = radius;
+      current_point.point_state = PointLabel::INIT;
+      current_point.orig_index = point_index;
 
-    // radial divisions
-    out_radial_ordered_points[radial_div].emplace_back(current_point);
-    ++point_index;
+      // radial divisions
+      out_radial_ordered_points[radial_div].emplace_back(current_point);
+      ++point_index;
+    }
   }
-  // sort by distance
-  for (size_t i = 0; i < radial_dividers_num_; ++i) {
-    std::sort(
-      out_radial_ordered_points[i].begin(), out_radial_ordered_points[i].end(),
-      [](const PointData & a, const PointData & b) { return a.radius < b.radius; });
+
+  {  // add scope for time keeper
```

Comment:
Same as above. Describe the role of the scope block.

Reply:
I change the comment to `scope for the timekeeper to track the time spent for sorting each horizontal angle group` .